libs/contrib/Data/Nat/Order/Strict.idr

```diff
@@ -1,25 +1,30 @@
 ||| Implementing `Decidable.Order.Strict` for `Data.Nat.LT`
 module Data.Nat.Order.Strict
 
-import Decidable.Order
+import Data.Nat
+import Control.Relation
+import Control.Order
 import Decidable.Order.Strict
 import Decidable.Equality
-import Data.Nat
 import Data.Nat.Order
 
-export
-irreflexiveLTE : (a : Nat) -> Not (a `LT` a)
-irreflexiveLTE 0     z_lt_z impossible
-irreflexiveLTE (S a) (LTESucc a_lt_a) = irreflexiveLTE a a_lt_a
+%default total
+
+public export
+Irreflexive Nat LT where
+  irreflexive {x = 0} _ impossible
+  irreflexive {x = S k} (LTESucc prf) =
+    -- why can't this be inlined?
+    let rec = irreflexive {x = k} {rel = LT} in
+      rec prf
```

Comment:
You can probably use `Data.Nat`'s  `view : (lt : LT m n) -> View lt` to make the definition cleaner.

Reply:
The same problem came up every way I tried with `view`. There's something funny going on here.

Reply:
This seems to work:

```idris
public export
Irreflexive Nat LT where
  irreflexive lt = case view lt of
    LTSucc lt => irreflexive {rel = Nat.LT} lt
```

For some reason, without the explicit qualifier Idris picks `Prelude.LT` over the `Nat` one.

Reply:
In fact, just putting in `Nat.LT` fixes it without any need for `view`.

Reply:
The view means you don't need to match on `x` and deal with an impossible case.